osu.Game/Screens/Play/GameplayClockContainer.cs

```diff
@@ -1,300 +1,126 @@
-        public readonly BindableBool IsPaused = new BindableBool();
-
         /// <summary>
-        /// The decoupled clock used for gameplay. Should be used for seeks and clock control.
+        /// The final clock which is exposed to gameplay components.
         /// </summary>
-        private readonly DecoupleableInterpolatingFramedClock adjustableClock;
-
-        private readonly double gameplayStartTime;
-        private readonly bool startAtGameplayStart;
-
-        private readonly double firstHitObjectTime;
-
-        public readonly BindableNumber<double> UserPlaybackRate = new BindableDouble(1)
-        {
-            Default = 1,
-            MinValue = 0.5,
-            MaxValue = 2,
-            Precision = 0.1,
-        };
+        public GameplayClock GameplayClock { get; private set; }
 
         /// <summary>
-        /// The final clock which is exposed to underlying components.
+        /// Whether gameplay is paused.
         /// </summary>
-        public GameplayClock GameplayClock => localGameplayClock;
-
-        [Cached(typeof(GameplayClock))]
-        private readonly LocalGameplayClock localGameplayClock;
-
-        private Bindable<double> userAudioOffset;
-
-        private readonly FramedOffsetClock userOffsetClock;
+        public readonly BindableBool IsPaused = new BindableBool();
 
-        private readonly FramedOffsetClock platformOffsetClock;
+        /// <summary>
+        /// The adjustable source clock used for gameplay. Should be used for seeks and clock control.
+        /// </summary>
+        protected readonly DecoupleableInterpolatingFramedClock AdjustableSource;
 
         /// <summary>
         /// Creates a new <see cref="GameplayClockContainer"/>.
         /// </summary>
-        /// <param name="beatmap">The beatmap being played.</param>
-        /// <param name="gameplayStartTime">The suggested time to start gameplay at.</param>
-        /// <param name="startAtGameplayStart">
-        /// Whether <paramref name="gameplayStartTime"/> should be used regardless of when storyboard events and hitobjects are supposed to start.
-        /// </param>
-        public GameplayClockContainer(WorkingBeatmap beatmap, double gameplayStartTime, bool startAtGameplayStart = false)
+        /// <param name="sourceClock">The source <see cref="IClock"/> used for timing.</param>
+        protected GameplayClockContainer(IClock sourceClock)
         {
-            this.beatmap = beatmap;
-            this.gameplayStartTime = gameplayStartTime;
-            this.startAtGameplayStart = startAtGameplayStart;
-            track = beatmap.Track;
-
-            firstHitObjectTime = beatmap.Beatmap.HitObjects.First().StartTime;
-
             RelativeSizeAxes = Axes.Both;
 
-            adjustableClock = new DecoupleableInterpolatingFramedClock { IsCoupled = false };
+            AdjustableSource = new DecoupleableInterpolatingFramedClock { IsCoupled = false };
+            AdjustableSource.ChangeSource(sourceClock);
```

Comment:
> Is this just done eagerly here instead on restart as before

Yep, that's exactly it.

Reply:
I've moved it back into `Restart()` (now named `Reset()`), because I noticed that changing the source in the ctor causes the track to be restarted in `PlayerLoader`,